test/alu_8bit_sbc_test.c

```diff
@@ -0,0 +1,248 @@
+    const uint8_t a = 128;
+    const uint8_t expected = 0;
+    set_flag(0, C_FLAG);
+
+    uint8_t actual = cpu_setup(opcode, address, a, a, &cpu.AF.bytes.high);
+
+    // Z should be set, as our result is 0
+    cr_expect(eq(u8, get_flag_bit(Z_FLAG), 1));
+
+    // N should be set
+    cr_expect(eq(u8, get_flag_bit(N_FLAG), 1));
+
+    // H should NOT be set because of no half-borrow
+    cr_expect(zero(u8, get_flag_bit(H_FLAG)));
+
+    // C should NOT be set because of no borrow
+    cr_expect(zero(u8, get_flag_bit(C_FLAG)));
+
+    // check if value is correct
+    cr_expect(eq(u8, actual, expected));
+
+    // check if PC is updated correctly
+    cr_expect(eq(u8, cpu.PC, (address + 1)));
+}
+
+Test(sbc_a_n, sbc_a_b_only_half_borrow, .init = cpu_mmu_setup, .fini = cpu_teardown) {
+    const uint8_t opcode = 0x98;
+    const uint8_t a = 0b0100111;
+    const uint8_t b = 0b0000111;
+    const uint8_t expected = a - b - 1;
+    const uint16_t address = rand() % (0xFFFF);
+    set_flag(1, C_FLAG);
+
+    uint8_t actual = cpu_setup(opcode, address, a, b, &cpu.BC.bytes.high);
+
+    // Z should not be set, as our result is not 0
+    cr_expect(zero(u8, get_flag_bit(Z_FLAG)));
+
+    // N should always be set
+    cr_expect(eq(u8, get_flag_bit(N_FLAG), 1));
+
+    // H should be set because borrow occured
+    cr_expect(eq(u8, get_flag_bit(H_FLAG), 1));
+
+    // C should not be set because of no borrow
+    cr_expect(zero(u8, get_flag_bit(C_FLAG)));
+
+    // check if value is correct
+    cr_expect(eq(u8, actual, expected));
+
+    // check if PC is updated correctly
+    cr_expect(eq(u8, cpu.PC, address + 1));
+}
+
+Test(sbc_a_n, sbc_a_c_borrow, .init = cpu_mmu_setup, .fini = cpu_teardown) {
+    const uint8_t opcode = 0x99;
+    const uint8_t a = 0b0000111;
+    const uint8_t c = 0b0010101;
+    const uint8_t expected = a - c - 1;
+    const uint16_t address = rand() % (0xFFFF);
+    set_flag(1, C_FLAG);
+    
+    uint8_t actual = cpu_setup(opcode, address, a, c, &cpu.BC.bytes.low);
+
+    // Z should not be set, as our result is not 0
+    cr_expect(zero(u8, get_flag_bit(Z_FLAG)));
+
+    // N should always be set
+    cr_expect(eq(u8, get_flag_bit(N_FLAG), 1));
+
+    // H should not be set because no HB occured
+    cr_expect(zero(u8, get_flag_bit(H_FLAG)));
+
+    // C should be set because of borrow
+    cr_expect(eq(u8, get_flag_bit(C_FLAG), 1));
+
+    // check if value is correct
+    cr_expect(eq(u8, actual, expected));
+
+    // check if PC is updated correctly
+    cr_expect(eq(u8, cpu.PC, address + 1));
+}
+
+Test(sbc_a_n, sbc_a_d_res_only, .init = cpu_mmu_setup, .fini = cpu_teardown) {
+    const uint8_t opcode = 0x9A;
+    const uint8_t a = rand() % (0xFF);
+    const uint8_t d = rand() % (0xFF);
+    const uint8_t expected = a - d - 1;
+    const uint16_t address = rand() % (0xFFFF);
+    set_flag(1, C_FLAG);
+
+    uint8_t actual = cpu_setup(opcode, address, a, d, &cpu.DE.bytes.high);
+
```

Comment:
I think that there are some flag checks missing here

Reply:
No, because the goal of the remaining tests is to simply check the correct value results from the arithmetic, as we checked the different flag configurations in the preceding test cases

Reply:
Ouh I see now, hence the name of the test, sorry for the comments, I didn't get that. It makes sense since this is necessary because of the randomization